spec/hedgelog_spec.rb

```diff
@@ -1,11 +1,300 @@
+  describe '#level_from_int' do
+    it 'returns a symbol for the level name, given the related integer' do
+      expect(logger.send(:level_from_int, 0)).to eq :debug
+    end
+    it 'returns a symbol if a symbol or string are passed in' do
+      expect(logger.send(:level_from_int, :debug)).to eq :debug
+      expect(logger.send(:level_from_int, 'debug')).to eq :debug
+    end
+  end
+
+  # This test is just for coverage
+  describe '#debugharder' do
+    let(:callinfo) { '' }
+
+    context 'when the path is out of the load path' do
+      it 'doesnt explode' do
+        expect(Hedgelog::BACKTRACE_RE).to receive(:match).with(callinfo) { [] }
+        expect($LOAD_PATH).to receive(:find) { false }
+        logger.send(:debugharder, callinfo)
+      end
+    end
+  end
+
+  describe '#channel' do
+    let(:channel) do
+      logger.channel('channel')
+    end
+    subject do
+      channel.debug 'Foo'
+      JSON.parse(log_dev.string)
+    end
+
+    it { should include('channel' => 'channel') }
+
+    context 'with context on the main channel' do
+      before :each do
+        logger[:c1] = 'test'
+      end
+      after :each do
+        logger.clear_channel_context
+      end
+
+      it 'includes the context under the "context" key' do
+        expect(subject['context']).to include('c1' => 'test')
+      end
+    end
+
+    context 'with context on the channel' do
+      before :each do
+        channel[:c2] = 'test'
+      end
+      after :each do
+        channel.clear_channel_context
+      end
+
+      it 'includes the context under the "context" key' do
+        expect(subject['context']).to include('c2' => 'test')
+      end
+    end
+
+    context 'when channels are nested' do
+      let(:channel) do
+        logger.channel('channel').channel('nested_channel')
+      end
+
+      it { should include('channel' => 'channel => nested_channel') }
+    end
+  end
+
+  describe 'performance' do
+    let(:log_dev) { '/dev/null' }
+    let(:standard_logger) { Logger.new(log_dev) }
+    let(:hedgelog_logger) { Hedgelog.new(log_dev) }
+
+    context 'when logging a string' do
+      let(:message) { 'log message' }
+
+      context 'when in debug mode' do
+        it 'is not be more than 8x slower than standard ruby logger' do
+          standard_benchmark = Benchmark.realtime { 1000.times { standard_logger.debug(message) } }
+          hedgelog_benchmark = Benchmark.realtime { 1000.times { hedgelog_logger.debug(message) } }
+
+          expect(hedgelog_benchmark).to be <= standard_benchmark * 8
+        end
+      end
+
+      context 'when not in debug mode' do
+        let(:standard_logger) do
+          logger = Logger.new(log_dev)
+          logger.level = Logger::INFO
+          logger
+        end
+        let(:hedgelog_logger) do
+          logger = Hedgelog.new(log_dev)
+          logger.level = Logger::INFO
+          logger
+        end
+
+        it 'is not be more than 4x slower than standard ruby logger' do
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.